src/main/webapp/app/entities/file-upload-exercise/file-upload-exercise.route.ts

```diff
@@ -26,54 +63,45 @@ export const fileUploadExerciseRoute: Routes = [
         canActivate: [UserRouteAccessService],
     },
     {
-        path: 'course/:courseId/file-upload-exercise',
-        component: FileUploadExerciseComponent,
-        data: {
-            authorities: ['ROLE_TA', 'ROLE_INSTRUCTOR', 'ROLE_ADMIN'],
-            pageTitle: 'artemisApp.textExercise.home.title',
+        path: 'course/:courseId/file-upload-exercise/new',
+        component: FileUploadExerciseUpdateComponent,
+        resolve: {
+            fileUploadExercise: FileUploadExerciseResolve,
         },
-        canActivate: [UserRouteAccessService],
-    },
-    {
-        path: 'course/:courseId/file-upload-exercise/:id',
-        component: FileUploadExerciseDetailComponent,
         data: {
             authorities: ['ROLE_TA', 'ROLE_INSTRUCTOR', 'ROLE_ADMIN'],
```

Comment:
I'm a bit confused. In the `FileUploadExerciseResource` you only authorized Instructors and Admins to create and update File Upload Exercises, but here the path is accessible for TAs as well? Is it intended this way?

Reply:
You are right here I changed the authorities